pkg/downloader/chart_downloader.go

```diff
@@ -158,6 +162,71 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+type inMemoryChart struct {
+	ChartData              *bytes.Buffer
+	ProvenanceVerification *provenance.Verification
+}
+
+// inMemoryCharts maps version to a particular chart
+type inMemoryCharts map[string]inMemoryChart
+
+// inMemoryChartList maps a chart name to a particular set of charts
+type inMemoryChartList map[string]inMemoryCharts
+
+var inMemoryCache = make(inMemoryChartList)
+
+//DownloadToInMemoryCache takes the chart reference and version and downloads it to the inMemoryCache for usage.
+//
+//The chart downloaded to the inMemoryCache can either be from the filesystem or from a URL
+func (c *ChartDownloader) DownloadToInMemoryCache(name, version string) (string, *provenance.Verification, error) {
```

Comment:
a general question: instead adding the new function and coping the most of the code from `DownloadTo` function, could we not instead modify the later so that does not write to the filesystem, e.g when `dest` arg is empty or `nil`? That would also allow CLI support immediately, by `HELM_REPOSITORY_CACHE` env cache to `""`

Reply:
Perhaps using a library like https://github.com/spf13/afero (or similar) could make things easier?

Reply:
I dont know about this, currently in Helm we follow the XDG MIME convention, this might interfere with that. @mattfarina Can you comment on this if possible ?